src/renderer/components/AgentList.vue

```diff
@@ -281,23 +295,72 @@ export default {
         console.error("Encountered error while attempting to establish mediation:", err)
       }
     },
+
     async process_mediator_invitation() {
       let connection = await this.new_agent_invitation_process(this, this.new_mediator_invitation);
       connection.unpacked_processor = this.mediatorInbound(connection);
       await this.send_mediation_request(connection);
       this.mediatorConnect(connection);
       this.new_mediator_invitation = "";
     },
+
+    getUrlVars: function(url) {
+        var vars = {};
+        var parts = url.replace(/[?&]+([^=&]+)=([^&]*)/gi, function(m,key,value) {
+          vars[key] = value;
+        });
+        return vars;
+      },
+
     async connect_clicked() {
-      this.invitation_error = "";
-      try {
-        await ConnectionsProtocol.new_agent_invitation_process(this, this.new_agent_invitation);
-      } catch (err) {
-        console.log("request post err", err);
-        this.invitation_error = err.message;
+      let invite_b64 = "";
+      if (this.new_agent_invitation.includes("?oob")) {
+        //OOB Invitation
+        invite_b64 = this.getUrlVars(this.new_agent_invitation)["oob"];
+        //base 64 decode
+        let invite_string = base64_decode(invite_b64);
+        let invite = JSON.parse(invite_string);
+        
+        this.invitation_error = "";
+        if (!invite.handshake_protocols || invite.handshake_protocols.length === 0) {
```

Comment:
I'm not a huge fan of this if-else-if-else chain. Perhaps consider using an object to associate handshake protocols strings with a handler:

```js
const protocolToHandler = {
    "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/connections/1.0": ConnectionsProtocol,
    "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/didexchange/1.0": DIDExProtocol,
    "https://didcomm.org/connections/1.0": ConnectionsProtocol,
    "https://didcomm.org/didexchange/1.0": DIDExProtocol,
}

let handled = false

for (const protocol of invite.handshake_protocols) {
    if (protocol in protocolToHandler) {
        await protocolToHandler[protocol].new_agent_invitation_process(this, invite)
        handled = true
        break
    }
}

if (!handled) {
    throw new Error("No supported handshake_protocols supplied")
}
```

Reply:
This would also give us "built-in" support for processing in priority order.